src/predicates.jl

```diff
@@ -108,10 +108,18 @@ end
 Base.isnan(x::ArbOrRef) = isnan(midref(x))
 Base.isnan(x::AcbOrRef) = isnan(midref(realref(x))) || isnan(midref(imagref(x)))
 
-Base.isnan(p::Union{ArbPoly,ArbSeries,AcbPoly,AcbSeries}) =
-    any(isnan, @inbounds p[i] for i = 0:degree(p))
-Base.isfinite(p::Union{ArbPoly,ArbSeries,AcbPoly,AcbSeries}) =
-    all(isfinite, @inbounds p[i] for i = 0:degree(p))
+function Base.isnan(p::Union{ArbPoly,ArbSeries,AcbPoly,AcbSeries})
+    @inbounds for i = 0:degree(cstruct(p))
```

Comment:
hmm, why is this `cstruct` here? is `degree(p)` not defined/gives different result?
(simple in-line comment would clarify this)

Reply:
Should have added a comment from the beginning! It's a microoptimization that avoids iterating over coefficients known to be zero for series. It would be more natural to write `degree(p.poly)` for series, but then that doesn't work for polynomials. Added a comment now!